packages/plugin-pnpm/sources/PnpmLinker.ts

```diff
@@ -254,9 +320,12 @@ function getStoreLocation(project: Project) {
   return ppath.join(getNodeModulesLocation(project), `.store` as Filename);
 }
 
-function getPackageLocation(locator: Locator, {project}: {project: Project}) {
+function getPackageLocation(locator: Locator, {project, createSelfReference}: {project: Project, createSelfReference: boolean}) {
   const pkgKey = structUtils.slugifyLocator(locator);
-  const pkgPath = ppath.join(getStoreLocation(project), pkgKey);
+  const prefixPath = createSelfReference
+    ? structUtils.getIdentVendorPath(locator)
+    : PortablePath.dot;
+  const pkgPath = ppath.join(getStoreLocation(project), pkgKey, prefixPath);
```

Comment:
Why put them in different location depending on whether there's a self-reference or not? Since we'll create a symlink in `.store/<hash>/n_m/<ident>/n_m/<ident>` it shouldn't be needed (if it's only an optimization, I'd favor consistency and simplicity, and use the same path format everywhere).